### PR TITLE
Single node deployment strategy

### PR DESCRIPTION
**Pull request by jhamman**
We have found that using Dask-kubernetes and prefect together on prefect-cloud is too unstable for our flows. This PR simplifies our runtime to simply deploy a local dask cluster on a single large node. This should allow us to use prefect cloud without the risk/overhead of a full distributed kubernetes cluster.

### CODE REVIEW

File: flows/methods/gard/flow.py [hunk 1]
```diff
@@ -140,10 +146,12 @@
...
         # )
         p['monthly_pyramid_path'] = pyramid(
-            p['full_space_model_output_path'], weights_pyramid_path=p['pyramid_weights'], levels=4
+            p['monthly_summary_full_space_path'],
+            weights_pyramid_path=p['pyramid_weights'],
+            levels=4,
         )
         p['annual_pyramid_path'] = pyramid(
-            p['full_space_model_output_path'], weights_pyramid_path=p['pyramid_weights'], levels=4
+            p['annual_summary_full_space_path'], weights_pyramid_path=p['pyramid_weights'], levels=4
```
Comment (jhamman) [inline flows/methods/gard/flow.py:154]: These changes correct what I assume was a bug. Please review / confirm.

File: cmip6_downscaling/methods/common/tasks.py [hunk 2]
```diff
@@ -158,7 +164,7 @@ def rechunk(
     path: UPath,
     pattern: str = None,
     template: UPath = None,
-    max_mem: str = "2GB",
+    max_mem: str = "5GB",
```
Comment (norlandrhagen) [inline cmip6_downscaling/methods/common/tasks.py:167]: Does the new zarr encoding raise the rechunker mem limit?

Comment (jhamman) [reply cmip6_downscaling/methods/common/tasks.py:167]: Yes. This cuts down on the number of intermediate chunks we have to generate.

**Comment by norlandrhagen** [comment]
Ran into the same task failures on `fit_and_predict` for both gard and bcsd for a cloud deployment. Also failed on a pangeo run. @jhamman 

```Task 'fit_and_predict': Exception encountered during task execution!
Traceback (most recent call last):
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/prefect/engine/task_runner.py", line 876, in get_task_run_state
    value = prefect.utilities.executors.run_task_with_timeout(
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/prefect/utilities/executors.py", line 467, in run_task_with_timeout
    return task.run(*args, **kwargs)  # type: ignore
  File "/home/jovyan/carbonplan/cmip6-downscaling/cmip6_downscaling/methods/bcsd/tasks.py", line 199, in fit_and_predict
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/xarray/core/dataset.py", line 2036, in to_zarr
    return to_zarr(
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/xarray/backends/api.py", line 1432, in to_zarr
    writes = writer.sync(compute=compute)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/xarray/backends/common.py", line 166, in sync
    delayed_store = da.store(
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/dask/array/core.py", line 1175, in store
    compute_as_if_collection(Array, store_dsk, map_keys, **kwargs)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/dask/base.py", line 319, in compute_as_if_collection
    return schedule(dsk2, keys, **kwargs)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/distributed/client.py", line 3018, in get
    results = self.gather(packed, asynchronous=asynchronous, direct=direct)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/distributed/client.py", line 2171, in gather
    return self.sync(
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/distributed/utils.py", line 309, in sync
    return sync(
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/distributed/utils.py", line 376, in sync
    raise exc.with_traceback(tb)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/distributed/utils.py", line 349, in f
    result = yield future
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/tornado/gen.py", line 762, in run
    value = future.result()
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/distributed/client.py", line 2034, in _gather
    raise exception.with_traceback(traceback)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/xarray/core/parallel.py", line 285, in _wrapper
    result = func(*converted_args, **kwargs)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/cmip6_downscaling/methods/bcsd/tasks.py", line 125, in _fit_and_predict_wrapper
    pointwise_model.fit(xtrain[run_parameters.variable], ytrain[run_parameters.variable])
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/skdownscale/pointwise_models/core.py", line 228, in fit
    self._models = _fit_wrapper(X, *args, **kws)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/skdownscale/pointwise_models/core.py", line 76, in _fit_wrapper
    scalar_obj[:] = [mod.fit(xdf, ydf, **kwargs)]
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/skdownscale/pointwise_models/bcsd.py", line 230, in fit
    X, y = self._validate_data(X, y, y_numeric=True)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/skdownscale/pointwise_models/base.py", line 81, in _validate_data
    X, y = self._check_X_y(X, y, **check_params)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/skdownscale/pointwise_models/base.py", line 13, in _check_X_y
    check_X_y(X, y)  # this may be inefficient
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/sklearn/utils/validation.py", line 979, in check_X_y
    y = _check_y(y, multi_output=multi_output, y_numeric=y_numeric)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/sklearn/utils/validation.py", line 994, in _check_y
    _assert_all_finite(y)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/sklearn/utils/validation.py", line 114, in _assert_all_finite
    raise ValueError(
ValueError: Input contains NaN, infinity or a value too large for dtype('float32').
```

**Comment by jhamman** [comment]
@norlandrhagen - what happens if you turn caching off?

**Comment by norlandrhagen** [comment]
> @norlandrhagen - what happens if you turn caching off?

caching was off for both.

**Pull request merged**